### PR TITLE
Wrap Opencensus tracing in interface.

### PR DESCRIPTION
**Pull request by cschuet**

### CODE REVIEW

File: async_grpc/rpc_handler.h [hunk 1]
```diff
@@ -21,15 +21,20 @@
...
+#include "google/protobuf/message.h"
 #include "grpc++/grpc++.h"
 #if BUILD_TRACING
-#include "opencensus/trace/span.h"
+#include "async_grpc/opencensus_span.h"
 #endif
 
 namespace async_grpc {
 
+// extern thread_local Span* rpc_span;
```
Comment (sebastianklose) [inline async_grpc/rpc_handler.h]: remove lines 34-37?

File: async_grpc/opencensus_span.cc [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "async_grpc/opencensus_span.h"
+#include "async_grpc/common/make_unique.h"
```
Comment (sebastianklose) [inline async_grpc/opencensus_span.cc]: remove unused include

File: async_grpc/rpc_handler.h [hunk 3]
```diff
@@ -68,12 +73,14 @@ class RpcHandler : public RpcHandlerInterface {
 
 #if BUILD_TRACING
   RpcHandler()
-      : trace_span_(opencensus::trace::Span::StartSpan(
-            RpcServiceMethodConcept::MethodName())) {}
-  virtual ~RpcHandler() { trace_span_.End(); }
+      : span_(
+            OpencensusSpan::StartSpan(RpcServiceMethodConcept::MethodName())) {
+    rpc_span = span_.get();
```
Comment (sebastianklose) [inline async_grpc/rpc_handler.h]: unused --> remove.

**Review by sebastianklose — APPROVED** [review]

**Comment by cschuet** [comment]
@wally-the-cartographer merge

**Pull request merged**